sshd-scp/src/test/java/org/apache/sshd/client/scp/ScpTest.java

```diff
@@ -589,6 +589,53 @@ public void testScpNativeOnDirWithPattern() throws Exception {
         }
     }
 
+    @Test
+    public void testScpVirtualOnDirWithPattern() throws Exception {
+        Path remoteDir = Files.createTempDirectory("remote");
+        sshd.setFileSystemFactory(new VirtualFileSystemFactory(remoteDir));
+
```

Comment:
Please use `getTempTargetRelativeFile` (method inherited from `JUnitTestSupport`) - it makes sure that the file(s) reside under `target` folder which is cleaned when `mvn clean` is invoked. Your code leaves "orphan" directories and files in the user's `/tmp` folder.

Reply:
Ack.